include/seqan3/alignment/pairwise/edit_distance_algorithm.hpp

```diff
@@ -60,9 +81,42 @@ class edit_distance_algorithm
     {}
     //!}
 
+    /*!\brief Invokes the alignment computation for every indexed sequence pair contained in the given range.
+     * \tparam indexed_sequence_pairs_t The type of the range of the indexed sequence pairs; must model
+     *                                  seqan3::detail::indexed_sequence_pairs.
+     *
+     * \param[in] indexed_sequence_pairs The indexed sequence pairs to align.
+     *
+     * \returns A std::vector over seqan3::alignment_result.
+     *
+     * \details
+     *
+     * Computes for each contained sequence pair the respective alignment and returns all alignment results in a
+     * vector.
+     */
+    template <indexed_sequence_pair_range indexed_sequence_pairs_t>
+    constexpr auto operator()(indexed_sequence_pairs_t && indexed_sequence_pairs)
```

Comment:
In this sprint I thought we only add `start_idx, range` and then do

```c++
result_vector.push_back((*this)(++start_idx, get<0>(sequence_pair), get<1>(sequence_pair)));
```

The interface change that we give the id with the sequence should be discussed with the team.

Reply:
But the current interface of the two way executor supports this interface already, since it already provides indexed sequence pairs. Hence the code is easier to adapt. So from the current state of the implementation this would make more sense wouldn't it?

Reply:
Oh interesting. I wasn't aware that we already zipped the sequence pair and the index internally like 7 months ago. I thought this was a new idea when we discussed that. When it makes absolutely sense to just pass the complete range directly to the algorithm.

@smehringer As you can see [here](https://github.com/seqan/seqan3/blame/e6406b468e0d98befa1b6a142da4bfe3a5d4efd6/include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp#L223) this is quite old.

Reply:
We should then update the tasks, so that we just pass that range through.